### ISSUE TITLE
Cannot authorize USB Webcam/USB-Microphone access over AppImage

### ISSUE THREAD

**Issue by ichrispa** [wireapp/wire-desktop#2064]
Wire version: 3.5.2881
Operating system: Linux/OpenSUSE

What steps will reproduce the problem?

1. Download/Install built AppImage.
2. Start wire
3. Open Preferences -> Audio/Video

What is the expected result?
Wire asks for permissions to use the camera or microphone if not already authorized.

What is the actual result?
"Enable from your browser Preferences" redirects the user to a browser for authorizing Camera/Microphone access. Browser settings have no effect on AppImage container...

After a recent update to wire, the AppImage cannot be used for placing (video/audio) calls. My WebCam provides the microphone via USB and does not require v4lcompat. In the settings, I am asked to authorize usage of my camera or microphone via my browser, which has no effect on the AppImage.

Placing video calls using the USB Cameras microphone works just fine in various browsers (Firefox, Opera, Vivaldi, Chrome) using the webApp.

Thanks in advance for your support.

**Comment by alexanderbystander** [comment]
I have the same issue but on windows.
So far I've tried uninstalling the app, and removing it's files and folders in app data.
I also removed all drivers for older mics and cameras that wire might pickup unnecessarily.

After reinstalling and restarting, the issue still persists. Including using previous versions of the app.

**Comment by tildica** [comment]
Can confirm. The latest version of Wire for Linux (both Appimage and FlatPak, v3.5.2881) does not detect microphones. Tested with Linux Mint (Xfce and Cinnamon).
After downgrading to version 3.4.2879 (Appimage) and clearing "/.config/Wire/" my microphone is properly detected.

**Comment by ruxandra-valcu** [comment]
Can also confirm. Web app still thankfully has functioning AV.

**Comment by x80486** [comment]
It looks like it might be different from system to system; mine works as expected:

![image](https://user-images.githubusercontent.com/34633617/51945421-b9ebc080-23ec-11e9-85c2-b89431351aad.png)

I can only use the Flatpak version, the AppImage doesn't even run in Arch Linux without (a dangling) `gtk2`.

**Comment by nynhex** [comment]
@x80486 +1 on arch, same situation

**Comment by bitinerant** [comment]
Does this resolve the issue?
```
sudo snap install wire
snap connect wire:camera core:camera
```

**Comment by ichrispa** [comment]
Sry @bitinerant, I neither use nor intend to switch to snap containers, the issue only concerns the AppImage.

With recent updates, the behavior has become erratic - sometimes the webcam and mic just appear & work and sometimes permissions are missing.

In any case, I can no longer reproduce the error reliable nor am i able to provide a debug trace or log to help specify the problem in detail. Hence i will close the issue.

Feel free to reopen if you can provide the devs with more specific, reproducable behavior.

**Comment by OZub** [comment]
I can confirm that the issue is still there 100% of time on all my Ubuntu machines.  Web app in Chromium works fine.  "Native" version looks like this:
![image](https://user-images.githubusercontent.com/49483938/55879366-bf107d00-5b9e-11e9-89c4-ae2df7e29023.png)

No calls are possible.
Snap version does not fix it.

**Comment by ichrispa** [comment]
Hi @OZub,

if you can provide an kubuntu version along with the type/USBID of your camera and the precise version of wire, we might be able to provide the developers with a 'steps to reproduce' section. We could then reopen the issue as long as the error can be reproduced.

Best regards,

ichrispa

**Comment by OZub** [comment]
Hello,

Thanks for the prompt reply.
It was observed on both 16.04 and 17.10 with different mics/cameras present in the system. Currently the issue is resolved on the system with 16.04, but I can't reliably reproduce the way of resolving it. It was a mixture of deleting all possible .config, .local, .kde folders in home dir as well as deleting all possible tmp and auth files in various system locations. The discovery that led me to the idea that it can be resolved in this way was the fact that it started working normally when I created the test user and Wire worked there with clean home folder (it could not be resolved by only cleaning the old home though. It was fixed only when I started messing around with other system folders).
I still have the issue with my 17.10 so we can try to reproduce.
Here's cam and mic IDs from the system:
Bus 001 Device 004: ID 0d8c:013c C-Media Electronics, Inc. CM108 Audio Controller
Bus 001 Device 002: ID 0458:708c KYE Systems Corp. (Mouse Systems) Genius WideCam F100
Current Wire version is 3.0.2816. The problem has started some months ago right after clicking "upgrade wire" that popped up in the program. I don't remember which was the  exact version then.

The issue could not be resolved when trying to use git version (ran with "yarn start"), only switching the user/deleting configs.
